### ISSUE TITLE
Cannot build ruby with gcc@7.2.0

### ISSUE THREAD

**Issue by healther** [spack/spack#7130]
### Expected Result
An fail-free build and install of ruby

### Actual Result
```
==> Moving resource stage
        source : /wang/users/wv385/cluster_home/spack_failure/spack/var/spack/stage/resource-rubygems-updated-ssl-cert-z536hrmlzxen6vf7jvkfxtrsx4jhppzv/
        destination : /wang/users/wv385/cluster_home/spack_failure/spack/var/spack/stage/ruby-2.2.0-z536hrmlzxen6vf7jvkfxtrsx4jhppzv/ruby-2.2.0/rubygems-updated-ssl-cert
==> Error: OSError: [Errno 20] Not a directory: '/wang/users/wv385/cluster_home/spack_failure/spack/var/spack/stage/resource-rubygems-updated-ssl-cert-z536hrmlzxen6vf7jvkfxtrsx4jhppzv/'
OSError: OSError: [Errno 20] Not a directory: '/wang/users/wv385/cluster_home/spack_failure/spack/var/spack/stage/resource-rubygems-updated-ssl-cert-z536hrmlzxen6vf7jvkfxtrsx4jhppzv/'

 /wang/users/wv385/cluster_home/spack_failure/spack/lib/spack/spack/package.py:1007, in do_stage:
     3                 raise ValueError("Can only stage concrete packages.")
     4     
     5             self.do_fetch(mirror_only)     # this will create the stage
>> 6             self.stage.expand_archive()
     7     
     8             if not os.listdir(self.stage.path):
     9                 raise FetchError("Archive was empty for %s" % self.name)

See build log for details:
  /wang/users/wv385/cluster_home/spack_failure/spack/var/spack/stage/ruby-2.2.0-z536hrmlzxen6vf7jvkfxtrsx4jhppzv/ruby-2.2.0/spack-build.out
```

### Steps to reproduce the issue

```console
$ spack install gcc@7.2.0
$ spack compiler add $(spack location -i gcc@7.2.0)
$ spack install ruby %gcc@7.2.0
```

### Information on your system
`packages.yaml` defaults to `gcc@7.2.0`

I had another problem some time ago already and at that point introduced an `args.append('--disable-install-rdoc')`, though I never came around to make a PR out of this. I gather that the rdoc install should be fixed in newer ruby versions and only appears for newer gcc's anyway. 

But this error seems to be due to the newly introduced gem-installation (#6935)

**Comment by goxberry** [comment]
Does #7138 fix the problem?

**Comment by baberlevi** [comment]
I'm having the same issue, and as @healther  mentioned in #7138, it seems to be caused by attempting to move the resource stage, which is a symlink, but the move is treating it like a directory.

I get this issue regardless of compiler, and on both rhel & fedora.

I think I found the issue, and a have fix at #7417

**Comment by healther** [comment]
@baberlevi can you check whether #7387 already fixes this for you?
